plugins/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureBlobStore.java

```diff
@@ -438,8 +449,21 @@ private void executeMultipartUpload(String blobName, InputStream inputStream, lo
             final List<String> blockIds = new ArrayList<>(nbParts);
             for (int i = 0; i < nbParts; i++) {
                 final long length = i < nbParts - 1 ? partSize : lastPartSize;
-                final Flux<ByteBuffer> byteBufferFlux =
-                    convertStreamToByteBuffer(inputStream, length, DEFAULT_UPLOAD_BUFFERS_SIZE);
+                Flux<ByteBuffer> byteBufferFlux = convertStreamToByteBuffer(inputStream, length, DEFAULT_UPLOAD_BUFFERS_SIZE);
+                if (i == nbParts - 1) {
+                    byteBufferFlux.doOnComplete(() -> {
+                        try {
+                            if (inputStream.available() > 0) {
```

Comment:
I wonder if we should just drop this check altogether instead? It seems kind of weird to check that we drained the stream fully when we also give the size we want to write (and the check relies on the fact that the stream reports `available()` as well which is not guaranteed).

Maybe we should just drop this check and only check that `currentTotalLength == blobSize` at the end? I think I'd prefer that option and it simplifies things?

Reply:
I'll take your suggestion to remove the check, although I'm torn between the complexity it adds in the code vs the aid in troubleshooting, but maybe the existing check about `currentTotalLength > blobSize` is sufficient.